2.2_comb_logic.ipynb

```diff
@@ -425,34 +643,34 @@
     "\n",
     "  ???\n",
     "}\n",
-    "class ParameterizedAdderTester(c: ParameterizedAdder, saturate: Boolean) extends PeekPokeTester(c) {\n",
-    "  // 100 random tests\n",
-    "  val cycles = 100\n",
-    "  import scala.util.Random\n",
-    "  import scala.math.min\n",
-    "  for (i <- 0 until cycles) {\n",
-    "    val in_a = Random.nextInt(16)\n",
-    "    val in_b = Random.nextInt(16)\n",
-    "    poke(c.io.in_a, in_a)\n",
-    "    poke(c.io.in_b, in_b)\n",
+    "\n",
+    "for (saturate <- Seq(true, false)) {\n",
+    "  test(new ParameterizedAdder(saturate)) { c =>\n",
+    "    // 100 random tests\n",
+    "    val cycles = 100\n",
+    "    import scala.util.Random\n",
+    "    import scala.math.min\n",
+    "    for (i <- 0 until cycles) {\n",
+    "      val in_a = Random.nextInt(16)\n",
+    "      val in_b = Random.nextInt(16)\n",
+    "      c.io.in_a.poke(in_a.U)\n",
```

Comment:
This could be a good candidate for abstracting this poke-and-expect operation into a test helper function.